build-logic-settings/configuration-cache-compatibility/src/main/kotlin/gradlebuild.configuration-cache-compatibility.settings.gradle.kts

```diff
@@ -0,0 +1,128 @@
+                "publishKotlinDslPrecompiledScriptPluginsPluginMarkerMavenPublicationToTestRepository",
+                "publishPluginMavenPublicationToTestRepository",
+                "publishPluginsToTestRepository",
+            ) -> false
+
+            // Core tasks
+            task.name in listOf(
+                "components",
+                "dependantComponents",
+                "model",
+            ) -> true
+
+            task.name.startsWithAnyOf(
+                "publish",
+                "idea",
+            ) -> true
+
+            task is GradleBuild -> true
+
+            // gradle/gradle build tasks
+            task.name in listOf(
+                "updateInitPluginTemplateVersionFile",
+                "resolveAllDependencies",
+            ) -> true
+
+            task.name.endsWith("Wrapper") -> true
+            task.name in listOf("docs", "stageDocs", "serveDocs") -> true
+            task.name.startsWith("userguide") -> true
+            task.name == "samplesMultiPage" -> true
+            task.typeSimpleName() in listOf(
+                "KtsProjectGeneratorTask",
+                "JavaExecProjectGeneratorTask",
+                "JvmProjectGeneratorTask",
+                "NativeProjectGeneratorTask",
+                "MonolithicNativeProjectGeneratorTask",
+                "NativeProjectWithDepsGeneratorTask",
+                "CppMultiProjectGeneratorTask",
+                "BuildBuilderGenerator",
+                "PerformanceTest",
+                "BuildCommitDistribution",
+                "DetermineBaselines",
+            ) -> true
+
+            // Third parties tasks
+
+            // Publish plugin
+            task.name == "login" -> true
+
+            // Kotlin/JS
+            // https://youtrack.jetbrains.com/issue/KT-50881
+            task.name in listOf("generateExternals") -> true
+
+            // JMH plugin
+            task.name in listOf("jmh", "jmhJar", "jmhReport") -> true
+
+            // Gradle Doctor plugin
+            task.name in listOf(
+                "graph", "graphMain",
+                "projectGraphReport",
+                "ripples",
+                "aggregateAdvice",
+            ) -> true
+
+            task.name.startsWithAnyOf(
+                "advice",
+                "analyzeClassUsage",
+                "analyzeJar",
+                "constantUsageDetector",
+                "createVariantFiles",
+                "findUnusedProcs",
+                "generalsUsageDetector",
+                "importFinder",
+                "inlineMemberExtractor",
+                "locateDependencies",
+                "misusedDependencies",
+                "reason",
+                "redundantKaptCheck",
+                "redundantPluginAlert",
+            ) -> true
+
+            else -> false
+        }
+    }
+
+    gradle.lifecycle.beforeProject {
+        tasks.configureEach {
+            val task = this
+            if (unsupportedTasksPredicate(task)) {
```

Comment:
🚀 This will also parallelize the work of checking the predicate against tasks in each project.